overlord/snapstate/target.go

```diff
@@ -699,7 +706,17 @@ func (p *pathInstallGoal) toInstall(ctx context.Context, st *state.State, opts O
 		trackingChannel = snapst.TrackingChannel
 	}
 
-	channel, err := resolveChannel(p.instanceName, trackingChannel, p.revOpts.Channel, opts.DeviceCtx)
+	var channel string
+	switch {
+	case p.revOpts.Channel != "":
+		channel = p.revOpts.Channel
+	case si.Channel != "":
+		channel = si.Channel
+	case trackingChannel != "":
+		channel = trackingChannel
+	}
```

Comment:
nitpick: I wonder if this should be closer to the code that checks if si.Channel and revOpts.Channel match? or that check should be moved closer to this?

Reply:
With the changes introduced to `resolveChannel`, in my refactor PR, this block will be going away/changing anyways.